libs/langchain-community/src/vectorstores/astradb.ts

```diff
@@ -98,7 +124,12 @@ export class AstraDBVectorStore extends VectorStore {
       ...documents[idx].metadata,
     }));
 
-    await this.collection.insertMany(docs);
+    const chunkedDocs = chunkArray(docs, this.batchSize);
+    const batchCalls = chunkedDocs.map((chunk) =>
+      this.collection?.insertMany(chunk)
+    );
```

Comment:
`this.caller` should wrap the `this.collection?.insertMany` call. See [this file](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-openai/src/embeddings.ts#L258) for an example

Reply:
My bad on this one thanks for the catch